lib/vrl/stdlib/src/parse_ruby_hash.rs

```diff
@@ -232,9 +289,49 @@ mod tests {
         assert!(child.get("nope").unwrap().is_null());
     }
 
+    #[test]
+    fn test_parse_key_number() {
+        let result = parse(r#"{ 42 => "hello world" }"#).unwrap();
+        assert!(result.is_object());
+        let result = result.as_object().unwrap();
+        assert!(result.get("42").unwrap().is_bytes());
+    }
+
+    #[test]
+    fn test_parse_key_colon() {
+        let result = parse(r#"{ :colon => "hello world" }"#).unwrap();
+        assert!(result.is_object());
+        let result = result.as_object().unwrap();
+        assert!(result.get(":colon").unwrap().is_bytes());
+    }
+
+    #[test]
+    fn test_parse_underscore() {
+        let result = parse(r#"{ with_underscore => "hello world" }"#).unwrap();
+        assert!(result.is_object());
+        let result = result.as_object().unwrap();
+        assert!(result.get("with_underscore").unwrap().is_bytes());
+    }
+
+    #[test]
+    fn test_parse_dash() {
+        let result = parse(r#"{ "with-dash" => "foo" }"#).unwrap();
+        assert!(result.is_object());
+        let result = result.as_object().unwrap();
+        assert!(result.get("with-dash").unwrap().is_bytes());
+    }
+
+    #[test]
+    fn test_parse_quote() {
+        let result = parse(r#"{ "with'quote" => "and\"double\"quote" }"#).unwrap();
+        assert!(result.is_object());
+        let result = result.as_object().unwrap();
+        assert!(result.get("with'quote").unwrap().is_bytes());
```

Comment:
I think it would be worth asserting on the value as well here, instead of checking for `is_bytes` only. Especially in the case of `and"double"quote`, but I would also match the value in the other test cases.